ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/FieldInfo.java

```diff
@@ -31,22 +31,26 @@
 
   public enum FieldType {
     SYSTEM, // To be removed in the future. 0.9 saw this value no longer used.
-    KEY
+    KEY,
+    HEADER
   }
 
   private final String name;
   private final SchemaInfo schema;
   private final Optional<FieldType> type;
+  private final Optional<String> headerKey;
 
   @JsonCreator
   public FieldInfo(
       @JsonProperty(value = "name", required = true) final String name,
       @JsonProperty(value = "schema", required = true) final SchemaInfo schema,
-      @JsonProperty("fieldType") final Optional<FieldType> type
+      @JsonProperty("fieldType") final Optional<FieldType> type,
+      @JsonProperty("headerKey") final Optional<String> headerKey
```

Comment:
Seems this field is optional in different places. What if you create another constructor that defaults to Optional.empty()?

Reply:
Done